[JIT] Support partially specified sizes/strides in IRParser

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44113 [JIT] Support partially specified sizes/strides in IRParser**

Differential Revision: [D23508149](https://our.internmc.facebook.com/intern/diff/D23508149)

Commit:
IR parser now able to handle graphs of mixed sized and unsized ranks, but using magic number

[ghstack-poisoned]

Commit:
Update on "[JIT] Support partially specified sizes/strides in IRParser "




[ghstack-poisoned]

ZolotukhinM commented on aten/src/ATen/core/type.cpp:
```diff
@@ -940,17 +940,24 @@ TensorTypePtr TensorType::create(
-      sizes.concrete_sizes()->size() == strides.concrete_sizes()->size());
-  auto sprops = strides.concrete_sizes().has_value()
+
+  if(sizes.concrete_sizes().has_value()){
```
Nit: formatting seems to be off (no space before {).

ZolotukhinM commented on aten/src/ATen/core/type.cpp:
```diff
@@ -940,17 +940,24 @@ TensorTypePtr TensorType::create(
+    return TensorType::create(
       scalar_type, device, symbol_sizes, sprops, requires_grad, undefined);
+  } else {
+    // are unsized ranks 
```
Nit: unfinished comment?

ZolotukhinM commented on aten/src/ATen/core/type.cpp:
```diff
@@ -940,17 +940,24 @@ TensorTypePtr TensorType::create(
+    TORCH_INTERNAL_ASSERT(sizes.sizes() && sizes.size());
+    auto symbol_sizes = SymbolicShape(*sizes.sizes());
+    return TensorType::create(
+      scalar_type, device, symbol_sizes, VaryingShape<Stride>(*sizes.size()), requires_grad, undefined); 
```
Nit: might be worth adding a comment explaining that the strides would be all null, but there would be N of them.

ZolotukhinM reviewed (commented):
Excellent work! The only missing piece is tests, once you add them it'll be ready to land.

Commit:
Update on "[JIT] Support partially specified sizes/strides in IRParser "


Differential Revision: [D23508149](https://our.internmc.facebook.com/intern/diff/D23508149)

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
🚢

Commit:
Update on "[JIT] Support partially specified sizes/strides in IRParser "


Differential Revision: [D23508149](https://our.internmc.facebook.com/intern/diff/D23508149)

[ghstack-poisoned]

Commit:
Update on "[JIT] Support partially specified sizes/strides in IRParser "


Differential Revision: [D23508149](https://our.internmc.facebook.com/intern/diff/D23508149)

[ghstack-poisoned]
